### PR TITLE
Added samples to Strings.kt to showcase stdlib string functions

### PR DESCRIPTION
**Pull request by Samaritan1011001**

### CODE REVIEW

**Comment by Samaritan1011001** [comment]
Any update on this pull request?

File: .idea/runConfigurations/Generate_Compiler_Tests.xml [hunk 1]
```diff
@@ -18,7 +18,6 @@
       </option>
       <option name="vmOptions" value="" />
     </ExternalSystemSettings>
-    <extension name="coverage" enabled="false" merge="false" sample_coverage="true" runner="idea" />
```
Comment (ilya-g) [inline .idea/runConfigurations/Generate_Compiler_Tests.xml]: This change is not related to samples, better to revert

File: libraries/stdlib/samples/test/samples/text/strings.kt [hunk 2]
```diff
@@ -79,4 +79,196 @@ class Strings {
         assertPrints(byteArray.toString(charset), "Hello")
     }
 
+
+    //Indices is IntRange
+    @Sample
+    fun slice() {
+        assertPrints("abcdef".slice(IntRange(1,2)),"bc")
+        assertPrints("abcdef".slice(IntRange(0,0)),"a")
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/text/strings.kt]: Usually `IntRange` is constructed with the `..` operator: `1..2`

File: libraries/stdlib/samples/test/samples/text/strings.kt [hunk 3]
```diff
@@ -79,4 +79,196 @@ class Strings {
...
+    }
+
+    //Without the startIndex
+    @Sample
+    fun startsWith(){
+        assertPrints("abcd".startsWith("ab",true),"true")
+        assertPrints("abcd".startsWith("Ab",true),"true")
+        assertPrints("Abcd".startsWith("Ab",false),"true")
+        assertPrints("abcd".startsWith("ab",false),"true")
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/text/strings.kt]: Better to use named parameter to pass `ignoreCase` for clarity

File: libraries/stdlib/samples/test/samples/text/strings.kt [hunk 4]
```diff
@@ -79,4 +79,196 @@ class Strings {
...
+    }
+
+    //delimiter: Char
+
+    @Sample
+    fun substringBefore(){
+        assertPrints("abcd".substringBefore("c"),"ab")
+        assertPrints("abcd".substringBefore("8"),"abcd")
+        assertPrints("kotlinot".substringBefore("ot"),"k")
+        assertPrints("kotlin".substringBefore('k'),"")
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/text/strings.kt]: I think it'd be ok to merge first and third sample. 
Better to use some real use case to illustrate these functions, for example to take a part of a dot-delimitied identifier, or a slash-delimited path.

Comment (ilya-g) [reply libraries/stdlib/samples/test/samples/text/strings.kt]: Could you also show the usage of these functions with `missingDelimiterValue` parameter specified?

File: libraries/stdlib/samples/test/samples/text/strings.kt [hunk 5]
```diff
@@ -79,4 +79,196 @@ class Strings {
...
+    fun substringAfterLast(){
+        assertPrints("abcd".substringAfterLast("c"),"d")
+        assertPrints("abcd".substringAfterLast("8"),"abcd")
+        assertPrints("kotlinot".substringAfterLast("o"),"t")
+        assertPrints("kotlin".substringAfterLast('o'),"tlin")
+    }
+
+    @Sample
+    fun toBoolean(){
+        assertPrints("true".toBoolean(),"true")
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/text/strings.kt]: Please use `assertTrue`/`assertFalse` assertions instead

File: libraries/stdlib/samples/test/samples/text/strings.kt [hunk 6]
```diff
@@ -79,4 +79,196 @@ class Strings {
...
+        assertPrints("  a bcd  ".trimEnd(),"  a bcd")
+        assertPrints("  abcd  ".trimEnd(),"  abcd")
+
+    }
+
+    @Sample
+    fun trimStartString(){
+
+        //with no arguments
+        assertPrints("  a bcd  ".trimStart(),"a bcd  ")
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/text/strings.kt]: These trailing spaces will be invisible when printed.

**Comment by ilya-g** [comment]
@Samaritan1011001 sorry, I've made some review comments, but forgot to submit the review, so they all were left pending.

File: libraries/stdlib/samples/test/samples/text/strings.kt (hunk 2 above)
Comment (ilya-g) [reply libraries/stdlib/samples/test/samples/text/strings.kt]: Also in `1..2` range, the end index 2 can be confused with the range length which also happens to be 2.

**Comment by Samaritan1011001** [comment]
@ilya-g I have made the changes asked for. Please do check it. Also sorry for the late reply.

**Comment by ilya-g** [comment]
Good, but some problems are still unaddressed.

-  Please reformat samples according to the [Kotlin style guide](https://kotlinlang.org/docs/reference/coding-conventions.html) (Ctrl+Alt+L could help to do that)
- String-to-number conversion samples should be extracted to a separate class with samples.
- The second argument of `assertPrints` should be a literal string
- function and local variable names should be camel-cased
- trailing spaces in `trim*` samples will be invisible when printed

**Comment by Samaritan1011001** [comment]
Done.

**Comment by Samaritan1011001** [comment]
@ilya-g Do I need to make other changes?

**Comment by Samaritan1011001** [comment]
Are we done here? @ilya-g
